dev/deno.ts

```diff
@@ -600,11 +620,142 @@ const denoDoc: Fig.Subcommand = {
+        trigger: (newToken, oldToken) => {
+          return newToken.indexOf(".") !== oldToken.indexOf(".");
+        },
+
+        // If there's not dot, the query term is the whole string. If there is
+        // a dot, the query term is everything after it. This intentionally
+        // uses indexOf returning -1, since -1 + 1 = 0 (correct index to slice)
+        getQueryTerm: (token) => {
+          return token.slice(token.indexOf(".") + 1);
+        },
+
+        // This is quite a long function, but it's conceptually simple:
+        // 1. There's an array that will be filled with doc nodes.
+        //    a. If there is no dot in the filter token, that array will be
+        //       populated with the top level nodes.
+        //    b. If there is a dot, everything up to the dot becomes the name.
+        //       All nodes with that name are collected, and the array is
+        //       populated with their children.
+        // 2. Those nodes are filtered, and their names are added to a set.
+        // 3. Those names in that set are transformed into suggestions.
+        postProcess: (out, tokens) => {
+          // The output for `deno doc --json` is `DocNode[]` - the types:
+          // https://github.com/denoland/deno_doc/blob/dbf9e21/lib/types.d.ts
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          let docNodes: any[];
+          try {
+            docNodes = JSON.parse(out);
+            if (!Array.isArray(docNodes)) {
+              throw new Error(`Output data was JSON, but was not an array`);
+            }
+          } catch (err) {
+            console.error("Returning early due to error:", err);
+            return [];
+          }
+
+          const showPrivateSymbols = tokens.includes("--private");
+
+          if (docNodes.length === 0) return [];
+
+          // The final token has to be the filter, it's the only way this
+          // generator could have been invoked.
+          const filterToken = tokens[tokens.length - 1];
+          const firstDotIndex = filterToken.indexOf(".");
+
+          // Based on whether the user has typed a period, this will be
+          // populated with either the top level nodes or the children of
+          // whatever node the user has searched for.
+          const suggestNodes = [];
+
+          if (firstDotIndex === -1) {
+            suggestNodes.push(...docNodes);
+          } else {
+            const filterName = filterToken.slice(0, firstDotIndex);
+
+            // It's not uncommon that there'd be multiple occurrences of the
+            // same name with different values, eg. overloads and interface
+            // merging. Deno's builtin types actually do this with the `Deno`
+            // interface. Typically this will only be one or two nodes.
+            const foundNodes = docNodes.filter((node) => {
+              return node.name === filterName;
+            });
```

Comment:
We can simplify this a bit:
```suggestion
            const foundNodes = docNodes.filter(({ name }) => name === filterName);
```

Reply:
Personally I'd like that! However, that line goes over 80 characters, so the pre-commit formatter puts the condition on the line below. Reading `const foundNodes = docNodes.filter((node) =>` feels awkward, like I've been left hanging, waiting for an expression, a block... something.
Here's what I prefer:
a. The entire callback goes on the next line, which is what it was before (but it's awkward to read)
b. The callback starts on the same line as `.filter` and uses a block instead of an expression

Regardless of what we end up with, it'll be a purely stylistic choice - semantically it's identical. Lmk your thoughts!

Reply:
Looking at the recent commits, I find the new solution pretty clean - no need to wrap the callback to a new line while being a single-liner, that's sweet! We can keep it as it is right now.